functions/index.js

```diff
@@ -143,3 +143,49 @@ app.post('/', isAuthenticated ,function(req,response)
 });
 
 exports.signUp = functions.https.onRequest(app);
+
+
+
+//<--------Random Fact Generation---------->
+//Generates random fact about techspardha.
+exports.randomFact = functions.https.onRequest((request , response) => {
+  const numberOfLines = 8;
+  const randomIndex = Math.floor(Math.random() * numberOfLines);
+  database.ref('/facts/' + randomIndex).on('value',function(snapshot){
+    console.log(snapshot.val());
+    response.status(401).json({
+      message : snapshot.val()
+    });
+  });
+});
+
+
+//<------Returning the array of all the videos------>
+//Returns the array of videos containing title and url of a video.
+exports.getVideo = functions.https.onRequest((request , response) => {
+  let items = [];
+  database.ref('/videos').on('value', function(snapshot) {
+    snapshot.forEach(function(childSnapshot) {
+      items.push(childSnapshot.key,{
+        title : childSnapshot.val().title,
+        url : childSnapshot.val().url
+      });
+    });
+  });
+  response.status(401).json(items);
+}); 
+
+
+//<-----Adding query to database------->
+//only add newly asked query to the database.
+exports.addQuery = functions.https.onRequest((request,response)=>{
+  const query = request.query.text;
+  var newPostKey = admin.database().ref().child('queries').push().key;
```

Comment:
validate query if not null.
Need to add captcha validation.
Directly push into reference, why building a separate object?